examples/transforms/scale.rs

```diff
@@ -0,0 +1,57 @@
+use bevy::prelude::*;
+
+struct Scaling {
+    scale_axis: Vec3,
+}
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(scale_cube.system())
+        .run();
+}
+
+fn scale_cube(mut cubes: Query<(&mut Transform, &mut Scaling)>, timer: Res<Time>) {
+    for (mut transform, mut cube) in cubes.iter_mut() {
+        if transform.scale.length() >= 10.0 {
+            cube.scale_axis *= -1.0;
+        } else if transform.scale.length() < Vec3::ONE.length() {
+            // switch to next axis and expand
+            transform.scale = Vec3::ONE;
+            cube.scale_axis *= -1.0;
+            cube.scale_axis = Vec3::from((cube.scale_axis.z, cube.scale_axis.x, cube.scale_axis.y));
```

Comment:
Glam supports swizzling if you `use glam::swizzles::*` (or maybe `use bevy_math::swizzles::*`?) you should be able to do `cube.scale_axis.zxy()` here.